kv/etcd/store.go

```diff
@@ -188,16 +216,29 @@ func (c *client) Watch(key string) (kv.ValueWatch, error) {
 	return w, err
 }
 
-func (c *client) watchChan(key string) clientv3.WatchChan {
-	return c.watcher.Watch(
-		context.Background(),
-		c.opts.KeyFn()(key),
-		// periodically (appx every 10 mins) checks for the latest data
-		// with or without any update notification
-		clientv3.WithProgressNotify(),
-		// receive initial notification once the watch channel is created
-		clientv3.WithCreatedNotify(),
-	)
+func (c *client) watchChanWithTimeout(key string) (clientv3.WatchChan, error) {
+	doneCh := make(chan struct{})
+
+	var watchChan clientv3.WatchChan
+	go func() {
+		watchChan = c.watcher.Watch(
+			context.Background(),
```

Comment:
we can pass in a context here, but that means the watch will be ended every context.Timeout, I feel that's unnecessary when the watch in stable.

Yeah the watchChan is essentially a receive channel, there is no good way to clean it up.